### PR TITLE
Add eslint.

### PR DESCRIPTION
**Pull request by BigBlueHat**
- Add eslint.
- Add auto-fixes from eslint.
- Sort imports.
- Fix various long lines.
- Removed unused variable.
- Ignore long template literals.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
   "main": "index.js",
   "scripts": {
     "start": "http-server .",
-    "build": "webpack"
+    "build": "webpack",
+    "lint": "eslint ."
```
Comment (davidlehn) [inline package.json]:
```suggestion
    "lint": "eslint --ext .cjs,.js ."
```

File: .eslintrc.cjs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  ignorePatterns: ["dist/*", "webpack.config.js"],
+  extends: [
+    'eslint-config-digitalbazaar',
+    'eslint-config-digitalbazaar/jsdoc',
+    'digitalbazaar/module',
+  ],
+  parserOptions: {
+    ecmaVersion: 'latest',
+    sourceType: 'module'
+  },
```
Comment (davidlehn) [inline .eslintrc.cjs]: the default digitalbazaar rules set module type and env to es2022
```suggestion
```

File: .eslintrc.cjs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+module.exports = {
+  env: {
+    browser: true
+  },
+  ignorePatterns: ["dist/*", "webpack.config.js"],
+  extends: [
+    'eslint-config-digitalbazaar',
+    'eslint-config-digitalbazaar/jsdoc',
+    'digitalbazaar/module',
```
Comment (davidlehn) [inline .eslintrc.cjs]: Use uniform style.  If this is a module, package.json should have a type field.
```suggestion
    'digitalbazaar',
    'digitalbazaar/jsdoc',
    'digitalbazaar/module'
```

File: .eslintrc.cjs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  ],
+  parserOptions: {
+    ecmaVersion: 'latest',
+    sourceType: 'module'
+  },
+  rules: {
+    'jsdoc/check-examples': 0,
+    'max-len': ['error', {ignorePattern: '\\* SPDX-License-Identifier: ',
+      ignoreUrls: true, ignoreTemplateLiterals: true}]
```
Comment (davidlehn) [inline .eslintrc.cjs]: Most of this is in digitalbazaar config 5.2

**Review by davidlehn — COMMENTED** [review]
Does w3c have a particular style they prefer?  Or is it ok to use the digitalbazaar style for this?  I'm of course fine with the digitalbazaar ways.

**Comment by BigBlueHat** [comment]
> Does w3c have a particular style they prefer? Or is it ok to use the digitalbazaar style for this? I'm of course fine with the digitalbazaar ways.

We've been using our style for years and no one's complained. Additionally, the W3C receives loads of code donations and enforcing a single style across all of that surface area would be a fools errand. So...it's not a problem we need to address here.

**Comment by TallTed** [comment]
fwiw, [W3C Manual of Style](https://www.w3.org/Guide/manual-of-style/)

**Comment by davidlehn** [comment]
> fwiw, [W3C Manual of Style](https://www.w3.org/Guide/manual-of-style/)

That doesn't appear to cover preferred code styles.

File: package.json (hunk 1 above)
Comment (BigBlueHat) [reply package.json]: It's weird that's still not the default for eslint...but maybe that's changed in v9. Will fix.

**Comment by BigBlueHat** [comment]
@davidlehn added your tweaks. I did not change the package to `type: module` however, because the code isn't setup to load that way...and it's way more overhead to change it than is necessary at this point. The linting mostly passes now, and it's good enough to keep things on the rails for the present.

**Comment by BigBlueHat** [comment]
@davidlehn went ahead and merged this based on earlier review. Happy to keep tweaking...but can't keep being blocked by fiddly things (sorry).

**Comment by TallTed** [comment]
@davidlehn — It hasn't been updated since 2014, but maybe [this _Coding style_ wiki page](https://www.w3.org/wiki/Coding_style) helps.

**Pull request merged**